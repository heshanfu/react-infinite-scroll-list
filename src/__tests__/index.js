import React from 'react';
import { shallow } from 'enzyme';
import InfiniteList from '../index';

describe('<InfiniteList />', () => {
  const defaultProps = {
    containerHeight: '750px',
    isLoading: false,
    isEndReach: false,
    onThresholdReach: () => {},
  };

  it('expect to render with default props', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </InfiniteList>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render with given props', () => {
    const props = {
      ...defaultProps,
      className: 'custom-infinite-list',
      containerTagName: 'ul',
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <li>Child 1</li>
        <li>Child 2</li>
        <li>Child 3</li>
      </InfiniteList>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onThresholdReach with event as parameter', () => {
    const props = {
      ...defaultProps,
      onThresholdReach: jest.fn(),
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </InfiniteList>,
    );

    component
      .find('.infinite-list')
      .simulate('scroll', {
        target: {
          clientHeight: 750,
          scrollHeight: 1000,
          scrollTop: 250,
        },
      });

    expect(props.onThresholdReach).toHaveBeenCalledWith({
      target: {
        clientHeight: 750,
        scrollHeight: 1000,
        scrollTop: 250,
      },
    });
  });

  it('expect to call onThresholdReach if default threshold is reach', () => {
    const props = {
      ...defaultProps,
      onThresholdReach: jest.fn(),
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </InfiniteList>,
    );

    component
      .find('.infinite-list')
      .simulate('scroll', {
        target: {
          clientHeight: 750,
          scrollHeight: 1000,
          scrollTop: 250,
        },
      });

    expect(props.onThresholdReach).toHaveBeenCalledTimes(1);
  });

  it('expect to call onThresholdReach if given threshold is reach', () => {
    const props = {
      ...defaultProps,
      onThresholdReach: jest.fn(),
      threshold: 100,
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </InfiniteList>,
    );

    component
      .find('.infinite-list')
      .simulate('scroll', {
        target: {
          clientHeight: 750,
          scrollHeight: 1000,
          scrollTop: 150,
        },
      });

    expect(props.onThresholdReach).toHaveBeenCalledTimes(1);
  });

  it('expect to not call onThresholdReach if default threshold is not reach', () => {
    const props = {
      ...defaultProps,
      onThresholdReach: jest.fn(),
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </InfiniteList>,
    );

    component
      .find('.infinite-list')
      .simulate('scroll', {
        target: {
          clientHeight: 750,
          scrollHeight: 1000,
          scrollTop: 200,
        },
      });

    expect(props.onThresholdReach).not.toHaveBeenCalled();
  });

  it('expect to not call onThresholdReach if given threshold is not reach', () => {
    const props = {
      ...defaultProps,
      onThresholdReach: jest.fn(),
      threshold: 100,
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </InfiniteList>,
    );

    component
      .find('.infinite-list')
      .simulate('scroll', {
        target: {
          clientHeight: 750,
          scrollHeight: 1000,
          scrollTop: 100,
        },
      });

    expect(props.onThresholdReach).not.toHaveBeenCalled();
  });

  it('expect to not call onThresholdReach if loading is true', () => {
    const props = {
      ...defaultProps,
      onThresholdReach: jest.fn(),
      isLoading: true,
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </InfiniteList>,
    );

    component
      .find('.infinite-list')
      .simulate('scroll', {
        target: {
          clientHeight: 750,
          scrollHeight: 1000,
          scrollTop: 250,
        },
      });

    expect(props.onThresholdReach).not.toHaveBeenCalled();
  });

  it('expect to not call onThresholdReach if end is reach', () => {
    const props = {
      ...defaultProps,
      onThresholdReach: jest.fn(),
      isEndReach: true,
    };

    const component = shallow(
      <InfiniteList
        {...props}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </InfiniteList>,
    );

    component
      .find('.infinite-list')
      .simulate('scroll', {
        target: {
          clientHeight: 750,
          scrollHeight: 1000,
          scrollTop: 250,
        },
      });

    expect(props.onThresholdReach).not.toHaveBeenCalled();
  });
});
