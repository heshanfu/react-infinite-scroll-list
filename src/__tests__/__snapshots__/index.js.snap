// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfiniteList /> on Container expect to render 1`] = `
<div
  className="-risl-container"
>
  <div>
    Child 1
  </div>
  <div>
    Child 2
  </div>
  <div>
    Child 3
  </div>
  <div
    className="-risl-sentinel"
  />
</div>
`;

exports[`<InfiniteList /> on Container expect to render with custom containerClassName 1`] = `
<div
  className="custom-container"
>
  <div>
    Child 1
  </div>
  <div>
    Child 2
  </div>
  <div>
    Child 3
  </div>
  <div
    className="-risl-sentinel"
  />
</div>
`;

exports[`<InfiniteList /> on Container expect to render with custom sentinelClassName 1`] = `
<div
  className="-risl-container"
>
  <div>
    Child 1
  </div>
  <div>
    Child 2
  </div>
  <div>
    Child 3
  </div>
  <div
    className="custom-sentinel"
  />
</div>
`;

exports[`<InfiniteList /> on Container expect to render with custom tagName 1`] = `
<ul
  className="-risl-container"
>
  <div>
    Child 1
  </div>
  <div>
    Child 2
  </div>
  <div>
    Child 3
  </div>
  <li
    className="-risl-sentinel"
  />
</ul>
`;

exports[`<InfiniteList /> on Container expect to render with threshold 1`] = `
<div
  className="-risl-container"
>
  <div>
    Child 1
  </div>
  <div>
    Child 2
  </div>
  <div>
    Child 3
  </div>
  <div
    className="-risl-sentinel"
  />
</div>
`;
